contracts/WorkspaceRegistry.sol

```diff
@@ -75,49 +81,96 @@ contract WorkspaceRegistry is Ownable, Pausable, IWorkspaceRegistry {
     }
 
     /**
-     * @notice Add admin to a workspace, can be called by workspace admins
+     * @notice Update workspace members' roles, can be called by workspace admins
      * @param _id ID of target workspace
-     * @param _admins New admins for managing workspace
-     * @param _emails emails of admin. admin[0] has email [0]
+     * @param _members Members whose roles are to be updated
+     * @param _roles Roles to be updated
+     * @param _enabled Whether to enable or disable the role
+     * @param _emails Emails of members.
      */
-    function addWorkspaceAdmins(
+    function updateWorkspaceMembers(
         uint96 _id,
-        address[] memory _admins,
+        address[] memory _members,
+        uint8[] memory _roles,
+        bool[] memory _enabled,
         string[] memory _emails
-    ) external whenNotPaused onlyWorkspaceAdmin(_id) withinLimit(_admins.length) {
-        for (uint256 i = 0; i < _admins.length; i++) {
-            address adm = _admins[i];
-            workspaceAdmins[_id][adm] = true;
+    ) external whenNotPaused onlyWorkspaceAdmin(_id) withinLimit(_members.length) {
+        require(_members.length == _roles.length, "UpdateWorkspaceMembers: Parameters length mismatch");
+        require(_members.length == _enabled.length, "UpdateWorkspaceMembers: Parameters length mismatch");
+        require(_members.length == _emails.length, "UpdateWorkspaceMembers: Parameters length mismatch");
+        for (uint256 i = 0; i < _members.length; i++) {
```

Comment:
Members length is only 1000 at Max. Doesn't need 256 bits..
Sorry i know this is a carryforward from a prev commit

Reply:
If I keep uint16 and if the length of array exceeds the max value of uint16 then it overflows and incorrect value is passed to modifier. Can be less than 1000 hence passing the case and storing more than the limit.